iroh/src/discovery.rs

```diff
@@ -659,10 +658,19 @@ mod tests {
             let disco = ConcurrentDiscovery::from_services(vec![Box::new(disco1)]);
             new_endpoint(secret, disco).await
         };
-        let ep1_addr = NodeAddr::new(ep1.node_id());
         // wait for out address to be updated and thus published at least once
         ep1.node_addr().await?;
-        let res = ep2.connect(ep1_addr, TEST_ALPN).await;
+
+        // 10x faster test via a 3s idle timeout instead of the 30s default
+        let mut config = TransportConfig::default();
+        config.keep_alive_interval(Some(Duration::from_secs(1)));
+        config.max_idle_timeout(Some(IdleTimeout::try_from(Duration::from_secs(3))?));
+        let opts = ConnectOptions::new().with_transport_config(Arc::new(config));
+
+        let res = ep2
+            .connect_with_opts(ep1.node_id(), TEST_ALPN, opts)
+            .await? // -> Connecting works
+            .await; // -> Connection is expected to fail
         assert!(res.is_err());
```

Comment:
This test always relied on the connection timing out - so it would always run *at least* 30s.
We used to not have a way of customizing the transport config to reduce the idle timeout, but now we have. So now this test succeeds/fails after only about 3 seconds.